viewing_party/party.py

```diff
@@ -32,28 +32,23 @@ def watch_movie(user_data, title):
 
 def get_watched_avg_rating(user_data):
     watched_movies = user_data["watched"]
-    length_of_wathced = len(watched_movies)
-
-    if length_of_wathced == 0:
-        return 0.0
-    
     total_rating = 0.0
     for movie in watched_movies:
         total_rating += movie["rating"]
-    return total_rating/length_of_wathced
+    try:
+        return total_rating/len(watched_movies)
+    except ZeroDivisionError:
+        return total_rating

```

Comment:
Personally, I prefer the guard clause in this instance. My previous comment wasn't meant to imply that we _should_ change this to EAFP, just to think about how we could. But if we _did_ want to use exception handling, this is exactly how we could do so!